probe-rs/src/bin/probe-rs/cmd/run.rs

```diff
@@ -64,6 +66,47 @@ pub struct Cmd {
     pub(crate) rtt_scan_memory: bool,
 }
 
+/*
+ * For multi-core targets, it infers the target core from the RTT symbol.
+ */
+fn get_target_core_id(session: &Session, elf_file: &Path) -> Option<usize> {
+    let mut file = File::open(elf_file).ok()?;
+    let address = RttActiveTarget::get_rtt_symbol(&mut file)?;
+
+    tracing::debug!("RTT symbol found at 0x{:08x}", address);
+
+    let target_memory = session
+        .target()
+        .memory_map
+        .iter()
+        .filter_map(|region| {
+            if let MemoryRegion::Ram(region) = region {
+                Some(region)
+            } else {
+                None
+            }
+        })
+        .find(|region| region.range.contains(&address))?;
+
+    tracing::debug!("RTT symbol is in RAM region {:?}", target_memory.name);
+
+    let core_name = target_memory.cores.first()?;
+    let core_id = session
+        .target()
+        .cores
+        .iter()
+        .position(|core| core.name == *core_name)?;
+
+    tracing::debug!("RTT symbol is in core {}", core_id);
+
+    Some(core_id)
+}
+
+fn get_target_core<'a>(session: &'a mut Session, elf_file: &Path) -> Result<Core<'a>, Error> {
```

Comment:
IMO this function should return the core's ID, so the result can be reused. Reading through the .elf every time you need this is not something I'm happy with, even if we currently only call this function in two places.

Reply:
Thanks! I change that. The `get_rtt_symbol` still be called again [here](https://github.com/probe-rs/probe-rs/blob/4cd55afb4662fa9b7bc83ae11eea39256b50d866/probe-rs/src/bin/probe-rs/util/rtt.rs#L37). Should I call only once and pass the address as a parameter?